sklearn/datasets/tests/test_value_dropper.py

```diff
@@ -0,0 +1,229 @@
+import numpy as np
+
+from sklearn.datasets import ValueDropper
+from sklearn.datasets import make_classification, make_regression
+from sklearn.utils.testing import assert_equal
+from sklearn.utils.testing import assert_true
+from sklearn.utils.testing import assert_almost_equal
+from sklearn.utils.testing import assert_array_almost_equal
+from sklearn.utils.testing import assert_raise_message
+from sklearn.utils.testing import assert_raises_regexp
+from sklearn.preprocessing import LabelEncoder
+
+
+def test_value_dropper_mnar_clf():
+    # Test drop probabilites when missing distribution is
+    # given for classification problems
+    n_samples, n_features = 1000, 5
+    X, y = make_classification(n_samples=n_samples,
+                               n_classes=4,
+                               n_features=n_features,
+                               n_informative=5,
+                               n_redundant=0,
+                               n_repeated=0,
+                               random_state=0)
+    le = LabelEncoder().fit(['a', 'z', 'b', 'j'])
+    y_str = le.inverse_transform(y)
+    y_int = y
+
+    for y in (y_int, y_str):
+        classes = np.unique(y)
+
+        # Inplace dropping of values
+
+        # Samples from class 0 will have a drop-probability of 0.1
+        vd = ValueDropper(missing_proba={classes[0]: 0.1},
+                          missing_values=np.nan, random_state=0)
+        X_dropped = vd.transform(X, y)
+        missing_mask = np.isnan(X_dropped)
+
+        # Check the drop-probabilty for class 0
+        assert_almost_equal(missing_mask[y == classes[0]].sum() /
+                            float(np.sum(y == classes[0]) * n_features), 0.1,
+                            decimal=2)
+
+        # All the missing values are from y == 0
+        assert_almost_equal(
```

Comment:
What about comparing directly `y[missing_mask]` and `classes[0]`?
It combines this cryptic line and the next one